ReactAndroid/src/main/java/com/facebook/react/modules/appearance/AppearanceModule.java

```diff
@@ -66,6 +67,17 @@ private String colorSchemeForCurrentConfiguration(Context context) {
     return "light";
   }
 
+  @Override
+  public void setColorScheme(String style) {
+    if (style == "dark") {
```

Comment:
@birkir I missed that https://github.com/birkir/react-native/commit/e6338766ad0ac3b647d2d6f33a1ab5e4b3c7ae2e changed this from `.equals()` to `==`.

`==` against two Java strings does a reference comparison, instead of a value comparison. So it is not safe for using to compare string value.

Would you be willing to make a followup change to move this back to `.equals()`?

Reply:
Good spot and @birkir can you follow up with a test case to cover this please.

Reply:
Hey there @birkir @jacdebug. Because I think this change may not work, and we don't have any RNTester page or automated tests to verify a fix, I'm going to revert this change.

Do feel free to resubmit this change, with a fix, and test collateral. I do think this API makes sense as part of `Appearance`.

Reply:
> Hey there @birkir @jacdebug. Because I think this change may not work, and we don't have any RNTester page or automated tests to verify a fix, I'm going to revert this change

Hopefully we'll have it soon (cc @kelset :))

Reply:
Sure thing, added RNTester and fixed the bug in a new PR https://github.com/facebook/react-native/pull/36122

In retrospect, I think the strict equals bug came about because of coding style change (yoda) and the fact Kotlin uses `==` the same as Java `equals`.